src/state/markets/fifo.rs

```diff
@@ -581,22 +655,45 @@ impl<
     /// This function determines whether a PostOnly order crosses the book.
     /// If the order crosses the book, the function returns the price of the best order
     /// on the opposite side of the book in Ticks. Otherwise, it returns None.
-    fn check_for_cross(&mut self, side: Side, num_ticks: Ticks) -> Option<Ticks> {
-        let book = self.get_book_mut(side.opposite());
-        while let Some((o_id, order)) = book.get_min() {
-            let crosses = match side.opposite() {
-                Side::Bid => o_id.price_in_ticks >= num_ticks,
-                Side::Ask => o_id.price_in_ticks <= num_ticks,
-            };
-            if !crosses {
-                break;
-            } else if order.num_base_lots > BaseLots::ZERO {
-                return Some(o_id.price_in_ticks);
+    fn check_for_cross(
+        &mut self,
+        side: Side,
+        num_ticks: Ticks,
+        current_slot: u64,
+        current_unix_timestamp_in_seconds: u64,
+        record_event_fn: &mut dyn FnMut(MarketEvent<MarketTraderId>),
+    ) -> Option<Ticks> {
+        loop {
```

Comment:
why switch away from the `while let`?

Reply:
is this because you can't keep reference to `book` while calling `reduce_order_inner`?

Reply:
yeah mutable reference problems